server/src/main/java/com/linecorp/centraldogma/server/storage/StorageManager.java

```diff
@@ -81,6 +83,14 @@ default T create(String name, Author author) {
      */
     Set<String> listRemoved();
 
+    /**
+     * Returns a set of names for the elements which have been removed.
+     * The specified @{code filter} evaluates whether an {@link File} is collected or not
+     *
+     * @param filter the filter evaluates whether the {@link File} is collected or not
+     */
+    Set<String> listRemoved(Predicate<File> filter);
```

Comment:
This leaks the implementation detail of `DirectoryBasedStorageManager`. Not all storage may be filesystem-based. How about just retrieving the complete list using `listRemoved()` given that the list will not be very large?

Reply:
Oops! This API has been removed and added `listPurged()` to scan files which were `.purged`.